src/scripts/services/dimBungieService.factory.js

```diff
@@ -292,42 +212,18 @@ function BungieService($rootScope, $q, $timeout, $http, $state, dimState, toaste
     }
   }
 
-
-  /************************************************************************************************************************************/
-
   function getCharacters(platform) {
-    var data = {
-      token: null,
-      membershipId: null
-    };
-
-    var addTokenToData = assignResultAndForward.bind(null, data, 'token');
-    var getMembershipPB = getMembership.bind(null, platform);
+    const platformName = platform.type === 1 ? 'Xbox' : 'PSN';
 
-    var charactersPromise = getBungleToken()
-      .then(addTokenToData)
-      .then(getMembershipPB)
-      .then(function(membershipId) {
-        return getBnetCharactersRequest(data.token, platform, membershipId);
-      })
-      .then($http)
-      .then(handleErrors, handleErrors)
-      .then(processBnetCharactersRequest);
+    var charactersPromise = getMembership(platform)
+        .then((membershipId) => $http(bungieApiQuery(
+          `/Platform/Destiny/Tiger${platformName}/Account/${membershipId}/`
+        )))
+        .then(handleErrors, handleErrors)
+        .then(processBnetCharactersRequest);
 
     return charactersPromise;
 
-    function getBnetCharactersRequest(token, platform, membershipId) {
-      return {
-        method: 'GET',
-        url: 'https://www.bungie.net/Platform/Destiny/Tiger' + (platform.type === 1 ? 'Xbox' : 'PSN') + '/Account/' + membershipId + '/',
```

Comment:
You can remove the 'Tiger' trinary here and just use the platform.type (1 or 2) for Xbox and PSN.

https://destinydevs.github.io/BungieNetPlatform/docs/API-Test#/GetAccount/p:membershipType=1